lib/signex/HTTP.ex

```diff
@@ -0,0 +1,75 @@
+defmodule SignEx.HTTP do
+  @moduledoc """
+  Verify the integrity of HTTP requests.
+
+  `SignEx.HTTP` validates the integrity of HTTP requests by checking digest and signature.
+  The digest header contains the digest of the request body, useing a configurable hash algorithm.
+  The authorization header contains a signature and parameters that secure the request headers.
+
+  *NOTE the signature will sign a subset of the headers,
+  It is required for the digest header to be in this list so guarantee the integrity of the whole request.*
+
+  - [Specification for the contents of the digest header](https://tools.ietf.org/html/rfc3230)
+  - [Sepcification for signing HTTP Messages](https://tools.ietf.org/html/draft-cavage-http-signatures-05)
+
+  ## Examples
+
+      # iex> signature_string([{"date", "Tue, 07 Jun 2014 20:51:35 GMT"}])
+      # "date: Tue, 07 Jun 2014 20:51:35 GMT"
+  """
+  @default_headers [:date]
+
+  def digest_header_for(body) do
+    "SHA-256=" <> Base.encode64(digest_content(body))
+  end
+
+  def digest_content(body) do
+    :crypto.hash(:sha256, body)
+  end
+
+  def check_digest_header("SHA-256=" <> digest, message) do
```

Comment:
Seems fine for us to only use SHA-256, provided that it's clearly documented (and ideally the server responds with a useful error response when the requirement isn't met)